### DISCUSSION TITLE
Facing "java.lang.instrument ASSERTION FAILED ***: "!errorOutstanding" with message transform method call failed at s\open\src\java.instrument\share\native\libinstrument\JPLISAgent.c line: 884" when adding @Step("xyz")

### DISCUSSION THREAD

**Discussion by RashmiKalpeshChauhan** [allure-framework/allure2#2509] [Bug Reports]
Facing "java.lang.instrument ASSERTION FAILED ***: "!errorOutstanding" with message transform method call failed at s\open\src\java.instrument\share\native\libinstrument\JPLISAgent.c line: 884" when adding @Step("xyz")

  <maven.compiler.source>21</maven.compiler.source>
        <maven.compiler.target>21</maven.compiler.target>
		<allure.version>2.24.0</allure.version>
		<aspectj.version>1.9.22</aspectj.version>

**Comment by GauriDev03** [comment]
*** java.lang.instrument ASSERTION FAILED ***: "!errorOutstanding" with message transform method call failed at ./src/java.instrument/share/native/libinstrument/JPLISAgent.c line: 873

**Comment by douglasawh** [comment]
Is AppDynamics being used here? We reproduced this with AppDynamics.

**Comment by douglasawh** [comment]
looking at this a bit more closely, out error is also 884. I wonder if the 873 error is a different version of Java perhaps?

Also, I see Java 21, but what is the point version? We are on 21.0.6.

**Comment by skuznetsov-al** [comment]
Can you confirm AppDynamics agent version + full -javaagent list/order + JDK vendor/patch + the exception right before the assertion?
